packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1121,6 +1130,9 @@ mixin TextInputClient {
   /// Requests that this client perform the given action.
   void performAction(TextInputAction action);
 
+  /// Notify client about new content insertion from Android keyboard.
+  void insertContent(KeyboardInsertedContent content);
```

Comment:
Looks like Google tests are failing, and it's because of the addition of this method.  Anyone that `implements` TextInputClient will be broken.

One thing we should consider doing is to add an empty implementation here.

```dart
void insertContent(KeyboardInsertedContent content) {}
```

This is similar to showToolbar etc. below.  This will still break anyone using `implements`, but for those using `with` it won't be a breaking change.

The question is, do we want to break users, telling them that this is a mandatory method they must implement?  Or do we want this method to be added without breaking users that use `with`, potentially allowing them to be unaware it was added?

Regardless,  I think this change needs a [migration guide](https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes).

Reply:
I think that adding the empty implementation is the best route to go. This new method is only used on Android (at the moment, at least), and isn't critical to ensure a working TextField. 

I can work on a migration guide, but from the looks of it, that needs to be created after the PR is merged & landed? Hope I'm reading that right, please correct me if I'm wrong.

Reply:
Sounds good, I'm on board with adding the empty implementation.

The migration guide should be landed in the flutter/website repo after this PR is merged, but you can open a draft PR whenever you're ready.  Use the [template](https://github.com/flutter/website/blob/main/src/release/breaking-changes/template.md) from the guide I linked above.  [Here](https://github.com/flutter/website/blob/main/src/release/breaking-changes/scribble-text-input-client.md) is a migration guide that was added recently for the same reason as this PR, as an example

Reply:
I created a draft PR for the migration guide at https://github.com/flutter/website/pull/7692

Reply:
Reminder to follow up on that migration guide PR: https://github.com/flutter/website/pull/7692

Reply:
Followed up :)